hrreport/CRM/HRReport/Upgrader.php

```diff
@@ -247,21 +247,45 @@ public function upgrade_1404() {
       'CRM_HRReport_Form_Activity_HRAbsenceDates'
     ];
     $reports = civicrm_api3('OptionValue', 'get', [
-      'return' => ['name'],
+      'return' => ['label'],
       'option_group_id' => 'report_template',
       'name' => ['IN' => $templates],
     ]);
 
     if ($reports['count'] > 0) {
       $templateIds = array_column(array_values($reports['values']), 'id');
+      $templateLabels = array_column(array_values($reports['values']), 'label');
       foreach ($templateIds as $templateId) {
         civicrm_api3('OptionValue', 'delete', ['id' => $templateId]);
       }
 
-      $sql = "DELETE FROM `civicrm_managed` WHERE `entity_id` IN ('" . implode("','", $templateIds) . "')";
+      $sql = "DELETE FROM `civicrm_managed` WHERE `entity_type` = 'ReportTemplate'";
+      $sql .= " AND `name` IN ('" . implode(" Template','", $templateLabels) . " Template')";
       CRM_Core_DAO::executeQuery($sql);
     }
 
+    $this->_up1404RemoveReportInstance();
+
     return TRUE;
   }
+
+  /**
+   * Removes absence report instances
+   */
+  private function _up1404RemoveReportInstance() {
```

Comment:
Please never prefix private methods with an underscore. Prefixing a method in a upgrader class is only necessary when we are writing upgrader steps in traits, to avoid conflicts with methods in other traits. Since that is not the case here, you can just name the method as `removeReportInstances()`

Reply:
Noted. Thanks 👍